croaring/src/bitmap/serialization.rs

```diff
@@ -0,0 +1,121 @@
+use super::Bitmap;
+
+use std::ffi::{c_char, c_void};
+
+pub struct PortableSerializer {}
```

Comment:
Recommend using a trait here, something like

```rust
pub trait Serializer {
    fn serialize_into<'a>(bitmap: &Bitmap, dst: &'a mut Vec<u8>) -> &'a [u8];
    fn get_serialized_size_in_bytes(bitmap: &Bitmap) -> usize;
    fn try_deserialize(buffer: &[u8]) -> Option<Bitmap>;
}

// enum rather than struct, make it impossible to construct a value
// and unsuffixed, expected to be used as e.g `serialization::Portable`
pub enum Portable {}

impl Serializer for Portable { ... }
```

Also, for @saulius, we could use a breaking change here, and introduce something like

```rust
impl Bitmap {
    pub fn serialize_into<S: Serialize>(&self, dst: &'a mut Vec<u8>) -> &'a [u8] { ... }
    pub fn serialized_size<S: Serialize>(&self) -> usize { ... }
    pub fn try_deserialize<S: Serialize>(buffer: &[u8]) -> Option<Self> { ... }
}
```

We also could bring in `BitmapView` in a similar way

```rust
pub trait SerializeView {
    unsafe fn deserialize_view(data: &[u8]) -> BitmapView<'_>;
}

impl SerializeView for Portable { ... }
impl SerializeView for Frozen { ... }

impl<'a> BitmapView<'a> {
    pub unsafe fn new<S: SerializeView>(data: &'a [u8]) -> Self { ... }
}
````

Reply:
@Dr-Emann, thanks for weighing in 🙏🏻

I've tried implementing your suggestion in [61e8433](https://github.com/saulius/croaring-rs/pull/106/commits/61e8433621ce247b32cff9d94d9c64066922650c) -- is that what you had in mind? I decided to introduce three traits: `Serializer`, `Deserializer` and `ViewDeserializer`. This is to capture that `Frozen` only supports deserializing to a `BitmapView` and not to a `Bitmap`:

| Format | Serializer | Deserializer | ViewDeserializer |
|-|-|-|-
| `Portable` | ✅ | ✅ | ✅ |
| `Native` |  ✅ | ✅ | ❌ |
| `Frozen` |  ✅ | ❌ | ✅ |

Note that tests and documentation is still missing, but I was hoping to get the implementation right first.

Reply:
It's what I was thinking, yeah. @saulius, I personally think this is worth the breaking change (e.g. a call to `bitmap.get_serialized_size_in_bytes()` will no longer compile because it needs a Serializer type argument), but I'd like your opinion

Reply:
I also prefer this way to the way serialization is done in treemap, but we should probably make them consistent.

Reply:
This looks great, indeed.  I also agree with you that this is worth a breaking change. 

So far my line of thinking is this. Release current master as `0.9.0` - the last release under `saulius`. Then we can can merge this, work on treemap rework and transfer over to `RoaringBitmap` org (or the other way around). Newly refactored serialization interfaces can be a part of `1.0.0` released from under `RoaringBitmap`. How does that sound? If there are any other breaking-worth changes you both have in mind, that would also be a good time to incorporate.